storage/ndb/src/kernel/vm/mt.cpp

```diff
@@ -8013,13 +7989,29 @@ mt_get_instance_count(Uint32 block)
   return 0;
 }
 
+/**
+ * We map the blocks in the following manner.
+ * At first we have the LDM threads. Thus they start at thr_no = 0.
+ * There could be 0 LDM threads. If this is the case we have a setup
+ * with only receive threads. Thus the LDM blocks will still start
+ * from thr_no = 0.
+ *
+ * Next we have the TC threads, these are only available if they are
+ * specifically asked for in the thread configuration in config.ini.
+ *
+ * Next we have the receive threads. These are always present and they
+ * come after the LDM and tc threads.
+ *
+ * Finally at the end of the threads we have the main threads if there
+ * are any such setup.
+ */
 void
 mt_add_thr_map(Uint32 block, Uint32 instance)
 {
   Uint32 num_lqh_threads = globalData.ndbMtLqhThreads;
   Uint32 num_tc_threads = globalData.ndbMtTcThreads;
-  Uint32 thr_no = globalData.ndbMtMainThreads;
-  Uint32 num_restore_threads = globalData.ndbMtRecoverThreads;
+  Uint32 thr_no = 0;
+  Uint32 num_lqh_workers = num_lqh_threads;
```

Comment:
Can we directly use globalData.ndbMtLqhWorkers instead. No need to use a local variable num_lqh_workers, check and set it again below at L8039 and L8046.

Reply:
Good suggestion. Implemented it.